pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -0,0 +1,240 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package queuejob
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+
+	arbv1 "github.com/kubernetes-incubator/kube-arbitrator/pkg/apis/v1alpha1"
+	"github.com/kubernetes-incubator/kube-arbitrator/pkg/client"
+	"github.com/kubernetes-incubator/kube-arbitrator/pkg/client/clientset"
+	arbinformers "github.com/kubernetes-incubator/kube-arbitrator/pkg/client/informers"
+	informersv1 "github.com/kubernetes-incubator/kube-arbitrator/pkg/client/informers/v1"
+	listersv1 "github.com/kubernetes-incubator/kube-arbitrator/pkg/client/listers/v1"
+)
+
+const (
+	// QueueJobNameLabel label string for queuejob name
+	QueueJobNameLabel string = "xqueuejob-name"
+
+	// ControllerUIDLabel label string for queuejob controller uid
+	ControllerUIDLabel string = "controller-uid"
+)
+
+// controllerKind contains the schema.GroupVersionKind for this controller type.
+var controllerKind = arbv1.SchemeGroupVersion.WithKind("XQueueJob")
+
+//XController the XQueueJob Controller type
+type XController struct {
```

Comment:
My understand is that : this PR is only for client, so why XQueueJob controller here?

Reply:
It is a dummy controller that only listens for events